libs/components/src/next/form/datepicker/vl-datepicker.component.ts

```diff
@@ -0,0 +1,370 @@
+            appendTo: datepicker,
+        };
+
+        const options = {
+            ...staticOptions,
+            ...this.getDynamicOptions(),
+        } as unknown as Options;
+
+        Object.keys(options).forEach((key) => {
+            if (options[key as keyof Options] === undefined) delete options[key as keyof Options];
+        });
+        return options;
+    }
+
+    private getDatePicker() {
+        return this.shadowRoot?.querySelector('#wrapper') as HTMLDivElement;
+    }
+
+    private getFlatpickrWrapper() {
+        return this.shadowRoot?.querySelector('.flatpickr-wrapper') as HTMLDivElement;
+    }
+
+    private getFlatpickrCalendar() {
+        return this.shadowRoot?.querySelector('.flatpickr-calendar') as HTMLDivElement;
+    }
+
+    private getVisibleInputElement(): HTMLInputElement | undefined | null {
+        return this.renderRoot?.querySelector('input:not([type="hidden"])') as HTMLInputElement;
+    }
+
+    private updateOptionsForInstance(options: Options) {
+        Object.keys(options)
+            .map((key) => key as keyof Options)
+            .forEach((key) => {
+                this.instance?.set(key, options[key as keyof Options]);
+            });
+    }
+
+    private initializeComponent() {
+        if (this.getDatePicker() && !this.instance) {
+            this.instance = flatpickr(this.getDatePicker(), this.getOptions()) as unknown as Instance;
+            this.getDatePicker().classList.add('static');
+        }
+    }
+
+    private isInShadowDOM() {
+        return this.getRootNode() instanceof ShadowRoot;
+    }
+
+    onInput = (event: Event & { target: HTMLInputElement }) => {
+        const dateString = event?.target?.value;
+        if (this.pattern) {
+            if (!dateString.match(this.pattern)) {
+                this.updateValue(dateString);
+                return;
+            }
+        }
+        let parsedDate;
+        try {
+            parsedDate = flatpickr.parseDate(dateString, this.format);
+        } catch (error) {
+            this.updateValue(dateString);
+        } finally {
+            this.updateValue(dateString);
+            if (parsedDate instanceof Date && !isNaN(parsedDate as unknown as number)) {
+                this.instance?.setDate(dateString, false, this.format);
+            }
+        }
+    };
+
+    handleDateChange = (date: string) => {
+        const value = flatpickr.formatDate(new Date(date), this.format);
+        this.updateValue(value);
+    };
+
+    updateValue = (value: string) => {
+        this.value = value;
+        if (this.getVisibleInputElement() && value) this.getVisibleInputElement()!.value = value;
+        this.setValue(value);
+    };
+
+    getValue() {
+        return this.getVisibleInputElement()?.value;
+    }
+
+    get validationTarget(): HTMLInputElement | undefined | null {
+        return this.getVisibleInputElement();
+    }
+
+    /**
+     * Geeft focus aan het datepicker input element.
+     */
+    focus() {
```

Comment:
Kan deze private zijn?

Reply:
Deze kan niet private zijn gezien we transitief extenden van `HTMLElement`.